scripts/sexbound/plugins/climax/climax.lua

```diff
@@ -23,7 +24,7 @@ function Sexbound.Actor.Climax:new(parent, config)
         _scenarios = {},
         _soundEffects = {},
         _timer = {},
-        _inflation = 0,
+        _inflation = {},
```

Comment:
Why exactly are you instantiating variables as a table, when they later should be filled with a "class instance"? Assuming there ever be a case where the variable isn't properly filled later yet you still work with it, latest when you try to call a sub-function on it will you get a nil error.

Reply:
The short answer is: "I have no idea what I'm doing." ^^''

The lua doc told me there are no classes, only tables and weirder tables. So I thought declaring the internal variables as empty tables to be replaced by the instance later was the right thing to do.

This is exactly why this code needs review, sorry for the extra work. ;v;

Reply:
Well yes, Lua does not have classes or anything fancy at all. It has metatables, which is a weirder way to use old JavaScript prototypes.

However, strictly speaking defining a variable as an empty table and then replacing it with a different "funky" table just allocates unnecessary resources. And so, since it does not offer any failsave protection I'd say it's not necessary. Sure, reading values doesn't crash the script, as the table exists and any value read *from* the table is now "nil" instead of a "Trying to index nil value" error - but since it's supposed to be a "class instance" later down the line and 90% of all calls on it will be function calls, it will error regardless. Best to not initialize it at all and, if failsafety is needed, just do a quick `if self._inflation then ... end`